enhancements/node-healthcheck-operator.md

```diff
@@ -0,0 +1,435 @@
+Node Health Check Enhancement 
+
+---
+title: node-healthcheck-controller
+authors:
+  - "@rgolangh"
+reviewers:
+  - "@abeekof"
+  - "@n1r1"
+  - "@mshitrit"
+  - "@slintes"
+  - "@dhellmann"
+approvers:
+  - "@abeekof"
+  - "@n1r1"
+---
+
+# Node health checking
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs
+
+
+## Summary
+
+
+Enable opt in for automated node health checking and create objects
+for unhealthy nodes, to be handled later by external controllers
+that comply with the existing  [api].
+
+## Motivation
+
+Enable clusters that do not have a functioning Machine API to detect and respond to node-level failures in order to recover workloads , and/or restore cluster capacity.
+
+The existing Machine Health Check controller requires a functioning machine-api. This prevents its use in the majority of Kubernetes and OpenShift clusters.
+The proposed Node Healthcheck Controller (NHC) provides a generic solution that fits all installation types, together with an external remediation API that allows a pluggable remediation system to be installed and remediate unhealthy nodes.
+
+### Goals
+
+- Create an installation-type agnostic node health check solution
+
+- Feature parity with Machine Health Check controller
+
+- Providing a default strategy combined with [poison-pill].
+
+- Endorse usage of external [remediation api]
+
+- Allow defining different unhealthy criteria by node selectors, and have a sane strategy for handling nodes that match multiple NHC CRs.
+
+- Provide defaults health checks and default remedy custom resource creation.
+
+- Long term - Make Machine Health Check controller only implement the external remediation api and drop the default Machine API based remediation mechanism
```

Comment:
I think this step is optional and ultimately orthogonal to the proposal.

```suggestion
```

If we did head down this track, I believe it could be achieved without changing the MHC CRD.

Reply:
Seems like it belongs in the alternatives section. I don't think it needs to be a prerequisite for "provide a solution for non-MAPI clusters".

Reply:
If this is a long term desire then I think it belongs to the non-goal section for this proposal.